cmake/Modules/FindComputeCpp.cmake

```diff
@@ -51,146 +51,136 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
 mark_as_advanced(COMPUTECPP_USER_FLAGS)
 
-# Find OpenCL package
+set(COMPUTECPP_BITCODE "spir64" CACHE STRING
+  "Bitcode type to use as SYCL target in compute++")
+mark_as_advanced(COMPUTECPP_BITCODE)
+
 find_package(OpenCL REQUIRED)
 
 # Find ComputeCpp package
 
 # Try to read the environment variable
-if(DEFINED ENV{COMPUTECPP_PACKAGE_ROOT_DIR})
-  if(NOT COMPUTECPP_PACKAGE_ROOT_DIR)
-    set(COMPUTECPP_PACKAGE_ROOT_DIR $ENV{COMPUTECPP_PACKAGE_ROOT_DIR})
+if(DEFINED ENV{COMPUTECPP_DIR})
+  if(NOT ComputeCpp_DIR)
+    set(ComputeCpp_ROOT_DIR $ENV{COMPUTECPP_DIR} CACHE PATH
+      "The root of the ComputeCpp install")
   endif()
-endif()
-
-if(NOT COMPUTECPP_PACKAGE_ROOT_DIR)
-  message(FATAL_ERROR
-    "ComputeCpp package - Not found! (please set COMPUTECPP_PACKAGE_ROOT_DIR)")
 else()
-  message(STATUS "ComputeCpp package - Found")
+  set(ComputeCpp_ROOT_DIR ${ComputeCpp_DIR} CACHE PATH
+    "The root of the ComputeCpp install")
 endif()
 
-# Obtain the path to compute++
-find_program(COMPUTECPP_DEVICE_COMPILER compute++ PATHS
-  ${COMPUTECPP_PACKAGE_ROOT_DIR} PATH_SUFFIXES bin)
-if (EXISTS ${COMPUTECPP_DEVICE_COMPILER})
-  mark_as_advanced(COMPUTECPP_DEVICE_COMPILER)
-  message(STATUS "compute++ - Found")
-else()
-  message(FATAL_ERROR "compute++ - Not found! (${COMPUTECPP_DEVICE_COMPILER})")
-endif()
+find_program(ComputeCpp_DEVICE_COMPILER_EXECUTABLE compute++
+  PATHS ${ComputeCpp_ROOT_DIR}
+  PATH_SUFFIXES bin)
 
-# Obtain the path to computecpp_info
-find_program(COMPUTECPP_INFO_TOOL computecpp_info PATHS
-  ${COMPUTECPP_PACKAGE_ROOT_DIR} PATH_SUFFIXES bin)
-if (EXISTS ${COMPUTECPP_INFO_TOOL})
-  mark_as_advanced(${COMPUTECPP_INFO_TOOL})
-  message(STATUS "computecpp_info - Found")
-else()
-  message(FATAL_ERROR "computecpp_info - Not found! (${COMPUTECPP_INFO_TOOL})")
-endif()
+find_program(ComputeCpp_INFO_EXECUTABLE computecpp_info
+  PATHS ${ComputeCpp_ROOT_DIR}
+  PATH_SUFFIXES bin)
 
-# Obtain the path to the ComputeCpp runtime library
 find_library(COMPUTECPP_RUNTIME_LIBRARY
   NAMES ComputeCpp ComputeCpp_vs2015
-  PATHS ${COMPUTECPP_PACKAGE_ROOT_DIR}
-  HINTS ${COMPUTECPP_PACKAGE_ROOT_DIR}/lib PATH_SUFFIXES lib
-  DOC "ComputeCpp Runtime Library" NO_DEFAULT_PATH)
-
-if (EXISTS ${COMPUTECPP_RUNTIME_LIBRARY})
-  mark_as_advanced(COMPUTECPP_RUNTIME_LIBRARY)
-else()
-  message(FATAL_ERROR "ComputeCpp Runtime Library - Not found!")
-endif()
+  PATHS ${ComputeCpp_ROOT_DIR}
+  HINTS ${ComputeCpp_ROOT_DIR}/lib
+  PATH_SUFFIXES lib
+  DOC "ComputeCpp Runtime Library")
 
 find_library(COMPUTECPP_RUNTIME_LIBRARY_DEBUG
   NAMES ComputeCpp ComputeCpp_vs2015_d
-  PATHS ${COMPUTECPP_PACKAGE_ROOT_DIR}
-  HINTS ${COMPUTECPP_PACKAGE_ROOT_DIR}/lib PATH_SUFFIXES lib
-  DOC "ComputeCpp Debug Runtime Library" NO_DEFAULT_PATH)
-
-if (EXISTS ${COMPUTECPP_RUNTIME_LIBRARY_DEBUG})
-  mark_as_advanced(COMPUTECPP_RUNTIME_LIBRARY_DEBUG)
-else()
-  message(FATAL_ERROR "ComputeCpp Debug Runtime Library - Not found!")
+  PATHS ${ComputeCpp_ROOT_DIR}
```

Comment:
the `PATHS` in all `find_*()` calls are redundant.

Reply:
Yeah I'll stick to hints. Done.